monty-31/src/aarch64_neon/utils.rs

```diff
@@ -25,3 +28,86 @@ pub(crate) fn halve_neon<PMP: PackedMontyParameters>(input: uint32x4_t) -> uint3
         vhaddq_u32(input, to_add)
     }
 }
+
+/// Multiply a vector of Monty31 field elements by `2^{-N}`.
+///
+/// # Safety
+/// - The prime `P` must be of the form `P = r * 2^j + 1` with `r` odd and `j` equal to `TAD::TWO_ADICITY`.
+/// - `N` must be less `j`.
```

Comment:
Sorry, think this error was my fault.
```suggestion
/// - `N` must be less than or equal to `j`.
```